geotrek/trekking/tests/test_parsers.py

```diff
@@ -1148,6 +1150,34 @@ def test_trek_illustration_is_not_imported_on_missing_file_metadata(self, mocked
         self.assertEqual(Attachment.objects.count(), 0)
 
 
+class TestApidaeTrekParserConvertEncodingFiles(TestCase):
+    data_dir = "geotrek/trekking/tests/data/"
+
+    def test_fix_encoding_to_utf8(self):
+        file_name = f'{self.data_dir}/file_bad_encoding_tmp.kml'
+        copyfile(f'{self.data_dir}/file_bad_encoding.kml', file_name)
+
+        encoding = get_encoding_file(file_name)
+        self.assertNotEqual(encoding, "utf-8")
+
+        new_file_name = ApidaeTrekParser._maybe_fix_encoding_to_utf8(file_name)
+
+        encoding = get_encoding_file(new_file_name)
+        self.assertEqual(encoding, "utf-8")
+
+    def test_not_fix_encoding_to_utf8(self):
+        file_name = f'{self.data_dir}/file_good_encoding_tmp.kml'
+        copyfile(f'{self.data_dir}/file_good_encoding.kml', file_name)
+
+        encoding = get_encoding_file(file_name)
+        self.assertNotEqual(encoding, "utf-8")
```

Comment:
Given the name of your test this line should be `self.assertEqual(encoding, "utf-8")`.